base/pointer.jl

```diff
@@ -96,34 +96,168 @@ unsafe_wrap(Atype::Union{Type{Array},Type{Array{T}},Type{Array{T,N}}},
+end
 
 """
     unsafe_store!(p::Ptr{T}, x, i::Integer=1)
+    unsafe_store!(p::Ptr{T}, x, order::Symbol)
+    unsafe_store!(p::Ptr{T}, x, i::Integer, order::Symbol)
 
 Store a value of type `T` to the address of the `i`th element (1-indexed) starting at `p`.
-This is equivalent to the C expression `p[i-1] = x`.
+This is equivalent to the C expression `p[i-1] = x`. Optionally, an ordering can be
+provided.
 
 The `unsafe` prefix on this function indicates that no validation is performed on the
 pointer `p` to ensure that it is valid. Like C, the programmer is responsible for ensuring
 that referenced memory is not freed or garbage collected while invoking this function.
 Incorrect usage may segfault your program. Unlike C, storing memory region allocated as
 different type may be valid provided that that the types are compatible.
+
+This is compatible with storing pointers declared with `_Atomic` and `std::atomic`
+type in C11 and C++23 respectively. An error may be thrown if there is not support for
+atomically storing the Julia type `T`.
+
+!!! compat "Julia 1.10"
+     The `order` argument is available as of Julia 1.10.
 """
 unsafe_store!(p::Ptr{Any}, @nospecialize(x), i::Integer=1) = pointerset(p, x, Int(i), 1)
 unsafe_store!(p::Ptr{T}, x, i::Integer=1) where {T} = pointerset(p, convert(T,x), Int(i), 1)
+unsafe_store!(p::Ptr{T}, x, order::Symbol) where {T} = atomic_pointerset(p, x isa T ? x : convert(T,x), order)
+function unsafe_store!(p::Ptr, x, i::Integer, order::Symbol)
+    unsafe_store!(p + (aligned_sizeof(eltype(p)) * (Int(i) - 1)), x, order)
+end
+
+"""
+    unsafe_modify!(p::Ptr{T}, op, x, [order::Symbol]) -> Pair
+
+These atomically perform the operations to get and set a memory address after applying
+the function `op`. If supported by the hardware (for example, atomic increment), this may be
+optimized to the appropriate hardware instruction, otherwise its execution will be
+similar to:
+
+    y = unsafe_load(p)
+    z = op(y, x)
+    unsafe_store!(p, z)
+    return y => z
+
+The `unsafe` prefix on this function indicates that no validation is performed on the
+pointer `p` to ensure that it is valid. Like C, the programmer is responsible for ensuring
+that referenced memory is not freed or garbage collected while invoking this function.
+Incorrect usage may segfault your program.
+
+This is compatible with modifying pointers declared with `_Atomic` and `std::atomic`
+type in C11 and C++23 respectively. An error may be thrown if there is not support for
+atomically modifying the Julia type `T`.
+
+!!! compat "Julia 1.10"
+     This function requires at least Julia 1.10.
+
+See also: [`modifyfield!`](@ref), [`modifyproperty!`](@ref)
+"""
+function unsafe_modify!(p::Ptr, op, x, order::Symbol=:not_atomic)
+    return atomic_pointermodify(p, op, x, order)
+end
+
+"""
+    unsafe_replace!(p::Ptr{T}, expected, desired,
+                  [success_order::Symbol, [fail_order::Symbol=success_order]) -> (; old, success::Bool)
```

Comment:
```suggestion
                   [success_order::Symbol[, fail_order::Symbol=success_order]]) -> (; old, success::Bool)
```

Minor alignment/formatting :) Looking at `replacefield!`, seems like the second `]` is missing there as well :thinking: